pulsar-broker/src/test/java/com/yahoo/pulsar/broker/admin/AdminApiTest.java

```diff
@@ -377,6 +383,56 @@ public void brokers() throws Exception {
         admin.namespaces().deleteNamespace("prop-xyz/use/ns1");
         admin.clusters().deleteCluster("use");
         assertEquals(admin.clusters().getClusters(), Lists.newArrayList());
+        
+    }
+    
+    @Test
+    public void testDynamicConfiguration() throws Exception {
+        // create configuration znode
+        ZkUtils.createFullPathOptimistic(mockZookKeeper, BROKER_SERVICE_CONFIGURATION_PATH, "{}".getBytes(),
+                ZooDefs.Ids.OPEN_ACL_UNSAFE, CreateMode.PERSISTENT);
+        // Now, znode is created: set the watch and listener on the znode
+        Method getPermitZkNodeMethod = BrokerService.class.getDeclaredMethod("updateConfigurationAndRegisterListeners");
+        getPermitZkNodeMethod.setAccessible(true);
+        getPermitZkNodeMethod.invoke(pulsar.getBrokerService());
+
+        // (1) try to update dynamic field
+        final long shutdownTime = 10;
+        // update configuration
+        admin.brokers().updateConfiguration("brokerShutdownTimeoutMs", Long.toString(shutdownTime));
+        // verify value is updated
+        assertEquals(pulsar.getConfiguration().getBrokerShutdownTimeoutMs(), shutdownTime);
+
+        // (2) try to update non-dynamic field
+        try {
+            admin.brokers().updateConfiguration("zookeeperServers", "test-zk:1234");
+        } catch (Exception e) {
+            assertTrue(e instanceof PreconditionFailedException);
+        }
+
+        // (3) try to update non-existent field
+        try {
+            admin.brokers().updateConfiguration("test", Long.toString(shutdownTime));
+        } catch (Exception e) {
+            assertTrue(e instanceof PreconditionFailedException);
+        }
+
+    }
+    
+    @Test
+    public void testUpdateDynamicLocalConfiguration() throws Exception {
+
+        // (1) try to update dynamic field
+        final long shutdownTime = 10;
+        // update configuration
+        admin.brokers().updateConfiguration("brokerShutdownTimeoutMs", Long.toString(shutdownTime));
+        // Now, znode is created: updateConfigurationAndregisterListeners and check if configuration updated
+        Method getPermitZkNodeMethod = BrokerService.class.getDeclaredMethod("updateConfigurationAndRegisterListeners");
+        getPermitZkNodeMethod.setAccessible(true);
+        getPermitZkNodeMethod.invoke(pulsar.getBrokerService());
+        // verify value is updated
+        assertEquals(pulsar.getConfiguration().getBrokerShutdownTimeoutMs(), shutdownTime);
```

Comment:
I have added this logic in `testUpdateDynamicConfigurationWithZkWatch`.
This test-case directly triggers registerListener because pulsarService started when znode was not created and can't set the watch if node is not exist. so, in this testcase it explicitly triggers registerListener call.